builder_test.go

```diff
@@ -856,3 +857,49 @@ func TestRunWithMultiArg(t *testing.T) {
 	t.Logf("config: %#v", b.Config())
 	t.Logf(node.Dump())
 }
+
+func TestParseDockerignore(t *testing.T) {
+	dir, err := ioutil.TempDir("", "dockerignore*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	tests := []struct {
+		input, result []string
+	}{
+		{
+			input:  []string{"first", "second", "", "third", "fourth"},
+			result: []string{"first", "second", "third", "fourth"},
+		},
+		{
+			input:  []string{"", "first", "second", "", " #third", "#invalid pattern which shouldn't matter ("},
+			result: []string{"first", "second", " #third"},
+		},
+		{
+			input:  []string{"", "first", "second", "", "#third", ""},
+			result: []string{"first", "second"},
+		},
+	}
```

Comment:
Can you create an input that has a leading `#`?

Reply:
Do you mean on the first line?  The fifth line in this file has one.

Reply:
Yes, I think it would be good to have one on the first line too.

Reply:
Sure, adding it.

Reply:
One of the tests should have the `#` in the middle of a line (word)
```
                         input:  []string{"", "fi#rst", "#se#cond", "", " #third", "#invalid pattern which shouldn't matter ("},
			result: []string{"fi#rst", " #third"},
```

Reply:
Sure.  Added.